GetStream/stream-chat-android
# [3342] API call from token provider causing ANR issue

### 🎯 Goal

To avoid getting ANR during `connectUser` call

### 🛠 Implementation details

`setUser` is being invoked in the `IO` thread to invoke `TokenProvier.loadToken` off the `Main` thread.

### 🧪 Testing

- Put `Thread.sleep(10_000L)` into `ConstantTokenProvider.loadToken`
- build & launch any Sample app
- select any user
- make sure there is no `ANR` happening

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
